charts/opentelemetry-operator/release-checklist.md

```diff
@@ -0,0 +1,34 @@
+# Prerequisite
+
+- [ ] Make sure you have installed Kuttl which will be used to do the smoke tests later. See [Kuttl website](https://kuttl.dev/docs/)
+  for installation information.
+- [ ] Make sure you have cloned the [OpenTelemetry Operator](https://github.com/open-telemetry/opentelemetry-operator) in your workspace
+
+# Checklist
+
+- [ ] Change directory to `opentelemetry-helm-charts/charts/opentelemetry-operator`. `cd ./charts/opentelemetry-operator` should be helpful.
+- [ ] Run the command `go run main.go` to update the OTEL Collector CRD and get the latest tags of the two images which
+  the Operator will be using. Then update the corresponding values in the `values.yaml` (e.g., change the `manager.image.tag` value to `"v0.29.0"` or latest version).
+- [ ] Update `appVersion` value in the `Chart.yaml` if a new version of `manager.image.tag` is released
+- [ ] Download the latest OTEL Operator manifest from this [link](https://github.com/open-telemetry/opentelemetry-operator/releases/latest/download/opentelemetry-operator.yaml).
+  As you can see, there are several YAML files in this manifest separated by `---`.
+- [ ] Starting from the third YAML file, compare the YAML files in the manifest and the ones in the `charts/opentelemetry-operator/templates` directory.
+  The names of YAML files under `templates` directory are the same as the value of key `kind` of the YAML files in the manifest.
+- [ ] Update our template YAML files to maintain consistency with the ones in the manifest (especially be careful with `role.yaml` and `clusterrole.yaml`).
+  Create a new YAML file under `templates` directory if it doesn't exist.
+  Use `{{ template "opentelemetry-operator.name" . }}` to represent the name of OTEL Operator which probably is `opentelemetry-operator` in the manifest.
+  Use `{{ template "opentelemetry-operator.namespace" . }}` to represent the namespace which probably is `opentelemetry-operator-system` in the manifest.
+- [ ] Test the Operator Helm chart locally:
+  - [ ] Install cert-manager in your Kubernetes cluster: `kubectl apply -f https://github.com/jetstack/cert-manager/releases/download/v1.4.0/cert-manager.yaml`
+  - [ ] Install the Helm chart: `helm install my-op . -n opentelemetry-operator-system --create-namespace`
+  - [ ] Run the Helm tests: `helm test my-op -n opentelemetry-operator-system --debug`
+  - [ ] Change directory to the OpenTelemetry Operator
+  - [ ] Run the Kuttl smoke tests: `kubectl kuttl test ./tests/e2e`
+  - [ ] Make sure all the Helm tests and Kuttl smoke tests pass
```

Comment:
I would recommend to put the all the tests in the github actions test pipeline instead of the checklist. Not asking to do anything in this PR, maybe another github issue to track?

Reply:
It turns out our current GH actions ([helm/chart-testing](https://github.com/helm/chart-testing-action)) will run the `helm test` command (please see [here](https://github.com/helm/chart-testing/blob/main/doc/ct_install.md)). Then the maintainers don't need to run these tests but only KUTTL smoke tests.

KUTTL smoke tests sit in the [Operator repo](https://github.com/open-telemetry/opentelemetry-operator/blob/main/kuttl-test.yaml). I think maintainers must run the tests manually. Please let me know what you think. Thank you!

Reply:
Ideally all the tests should be moved to CI, so if any of them is failing a new PR cannot be merged. All the steps including fetching the kuttle tests from the operator repo are doable with github actions. Can we create a github issue for that?

Reply:
Yes, I will file another issue and implement this workflow. Thank you!